mm2src/adex_cli/src/scenarios/mm2_proc_mng.rs

```diff
@@ -0,0 +1,337 @@
+        .file_name()
+        .map_or("Undefined", |name: &OsStr| name.to_str().unwrap_or("Undefined"));
+
+    match daemon(true, true) {
+        Ok(Fork::Child) => {
+            command.output().expect("Failed to execute process");
+        },
+        Ok(Fork::Parent(pid)) => {
+            info!("Successfully started: {program:?}, forked pid: {pid}");
+        },
+        Err(error) => error!("Failed to fork a process: {error}"),
+    }
+}
+
+#[cfg(windows)]
+pub fn start_process_impl(mm2_binary: PathBuf) {
+    let Some(program) = mm2_binary.to_str() else {
+        error!("Failed to cast mm2_binary to &str");
+        return;
+    };
+    let program = match CString::new(program) {
+        Ok(program) => program,
+        Err(error) => {
+            error!("Failed to construct CString program path: {error}");
+            return;
+        },
+    };
+
+    let mut startup_info: STARTUPINFOA = unsafe { mem::zeroed() };
+    startup_info.cb = size_of::<STARTUPINFOA>() as u32;
+    let mut process_info: PROCESS_INFORMATION = unsafe { mem::zeroed() };
+
+    let result = unsafe {
+        CreateProcessA(
+            null(),
+            program.into_raw() as *mut i8,
+            std::ptr::null_mut(),
+            std::ptr::null_mut(),
+            0,
+            0,
+            std::ptr::null_mut(),
+            std::ptr::null(),
+            &mut startup_info,
+            &mut process_info,
+        )
+    };
+
+    match result {
+        0 => error!("Failed to start: {MM2_BINARY}"),
+        _ => info!("Successfully started: {MM2_BINARY}"),
+    }
+}
+
+#[cfg(all(unix, not(target_os = "macos")))]
+pub fn stop_process() {
+    let pids = find_proc_by_name(MM2_BINARY);
+    if pids.is_empty() {
+        info!("Process not found: {MM2_BINARY}");
+    }
+    pids.iter().map(u32::to_string).for_each(|pid| {
+        match Command::new(KILL_CMD)
+            .arg(&pid)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .status()
+        {
+            Ok(status) => {
+                if status.success() {
+                    info!("Process killed: {MM2_BINARY}:{pid}")
+                } else {
+                    error!("Failed to kill process: {MM2_BINARY}:{pid}")
+                }
+            },
+            Err(e) => error!("Failed to kill process: {MM2_BINARY}:{pid}. Error: {e}"),
+        };
+    });
+}
+
+#[cfg(windows)]
+pub fn stop_process() {
+    let processes = find_proc_by_name(MM2_BINARY);
+    for pid in processes {
+        info!("Terminate process: {}", pid);
+        unsafe {
+            let handy = OpenProcess(SYNCHRONIZE | PROCESS_TERMINATE, true as i32, pid);
+            TerminateProcess(handy, 1);
+        }
+    }
+}
+
+#[cfg(target_os = "macos")]
+pub fn start_process(mm2_cfg_file: &Option<String>, coins_file: &Option<String>, log_file: &Option<String>) {
+    let mm2_binary = match get_mm2_binary_path() {
+        Err(_) => return,
+        Ok(path) => path,
+    };
```

Comment:
let consider using `let else` here also to have a pattern 😄

Reply:
done